### PR TITLE
Harmony 1681 - Service Image Tag Endpoint

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1681

## Description
This PR adds the service-image-tag endpoint to allow providers to manage the deployed versions of their services.

## Local Test Steps
The GET actions (list) for this endpoint can use OAuth, but the PUT (update) action can only be run by passing in a bearer token, it will not redirect to EDL for authentication. This is by design
due to the limitations of redirects from EDL. So you must include a bearer token header in your requests when updating a tag.

There are four things to test:
1. list all services/tags(versions):
```
curl -Ln -bj " http://localhost:3000/service-image-tag
```
32 list a single service tag/version:
```
curl -Ln -bj "Authorization: Bearer <your token>"  http://localhost:3000/service-image-tag/harmony-service-example
```
43 Update a tag/version (doesn't really do anything yet except return a 201 if the user is an admin or in the deployers group and the tag validates.
```
curl  -X PUT -H 'Authorization: Bearer <your token>'  -H 'Content-type: application/json' http://localhost:3000/service-image-tag/harmony-service-example -d '{"tag": "foo"}'
```

You can test the verification that it doesn't allow changing the image host during updates by putting a ':' in your tag - it should get rejected

5. Check the documentation at http://localhost:3000/docs#service-image-tags-details

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [x] Documentation updated (if needed)

### CODE REVIEW

File: services/harmony/app/frontends/service-image-tags.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+ * Verify that the given tag is valid. Send an error if it is not.
+ * @param req - The request object
+ * @param res  - The response object - will be used to send an error if the validation fails
+ * @returns a Promise containing `true` if the tag is valid, false if not
+ */
+async function validateTag(
+  req: HarmonyRequest, res: Response,
+): Promise<boolean> {
+  const { tag } = req.body;
+  const tagRegex = /^[a-zA-Z\d_][a-zA-Z\d\-_.]{0,127}$/;
```
Comment (chris-durbin) [inline services/harmony/app/frontends/service-image-tags.ts:84]: I think you mentioned finding the tag requirements documented somewhere - maybe add a link to the documentation.

File: services/harmony/app/markdown/service-image-tags.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+### <a name="service-image-tags-details"></a> Managing Service Image Tags (Versions)
+
+Using the `service-image-tag` endpoint, service providers can manage the versions of their services deployed to an environment. Note that a user must be a member of either the EDL `Harmony Service Deployers`
+group or the EDL `Harmony Admin` group to access this endpoint, and requests to this endpoint _must_ include
+and EDL bearer token header, .e.g., `Authorization: Bearer <token>`.
```
Comment (chris-durbin) [inline services/harmony/app/markdown/service-image-tags.md:5]: an EDL

File: services/harmony/app/middleware/earthdata-login-oauth-authorizer.ts [hunk 3]
```diff
@@ -196,8 +196,10 @@ export default function buildEdlAuthorizer(paths: Array<string | RegExp> = []):
       } else if (token) {
         handler = handleAuthorized;
       } else if (requiresAuth) {
+        console.log('USING OAUTH');
```
Comment (chris-durbin) [inline services/harmony/app/middleware/earthdata-login-oauth-authorizer.ts:199]: Remove all the console.logs

**Review by chris-durbin — COMMENTED** [review]
Looks good - only a couple comments. I'll test it out locally before approving.

File: services/harmony/app/markdown/service-image-tags.md (hunk 2 above)
Comment (indiejames) [reply services/harmony/app/markdown/service-image-tags.md:5]: fixed

File: services/harmony/app/middleware/earthdata-login-oauth-authorizer.ts (hunk 3 above)
Comment (indiejames) [reply services/harmony/app/middleware/earthdata-login-oauth-authorizer.ts:199]: done

**Review by chris-durbin — APPROVED** [review]
Tested locally

File: services/harmony/app/frontends/service-image-tags.ts (hunk 1 above)
Comment (indiejames) [reply services/harmony/app/frontends/service-image-tags.ts:85]: done

**Review by ygliuvt — APPROVED** [review]

**Pull request merged**